src/hooks/swap/useSendSwapTransaction.tsx

```diff
@@ -129,12 +132,13 @@ export default function useSendSwapTransaction(
               throw new Error(t`Transaction rejected.`)
             } else {
               // otherwise, the error was unexpected and we need to convey that
-              console.error(`Swap failed`, error, address, calldata, value)
+              console.error(`Swap failed`, error, calldata, value)
+              onTxFail?.(error, error?.transactionHash)
```

Comment:
this isn't necessarily an ethereum tx fail issue -- it would throw here if we haven't formed a valid tx and therefore can't submit it

should we still call `onTxFail` here?